pbrd/pbr_nht.c

```diff
@@ -389,6 +397,7 @@ static afi_t pbr_nht_which_afi(struct nexthop_group nhg,
 
 	switch (nh_afi) {
 	case NEXTHOP_TYPE_IFINDEX:
+		install_afi = AFI_MAX;
```

Comment:
This nexthop-group produces this output:
Nexthop-Group: ZEP Table: 10003 Valid: 1 Installed: 1
	Valid: 1 nexthop 192.168.11.33
	Valid: 1 nexthop mveth3
robot# show ip route table 10003
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
       F - PBR, f - OpenFabric,
       > - selected route, * - FIB route, q - queued route, r - rejected route

F>* 0.0.0.0/0 [0/0] is directly connected, mveth3, 00:00:32
  *                 via 192.168.11.33, mveth1, 00:00:32
robot# show ipv6 route table 10003
Codes: K - kernel route, C - connected, S - static, R - RIPng,
       O - OSPFv3, I - IS-IS, B - BGP, N - NHRP, T - Table,
       v - VNC, V - VNC-Direct, A - Babel, D - SHARP, F - PBR,
       f - OpenFabric,
       > - selected route, * - FIB route, q - queued route, r - rejected route

F>r ::/0 [0/0] is directly connected, mveth3, 00:00:37
  r            via 192.168.11.33, mveth1, 00:00:37
robot# 

I think we need to only use AFI_MAX if all nexthops in nhg are NEXTHOP_TYPE_IFINDEX?

Reply:
Good catch, will do that.